tests/unit/clients/python/test_client.py

```diff
@@ -216,3 +287,183 @@ def test_deployment_sync_client(mocker, use_stream):
+):
+    random_port = port_generator()
+    on_always_mock = mocker.Mock()
+    on_error_mock = mocker.Mock()
+    on_done_mock = mocker.Mock()
+    client = Client(host=f'{protocol}://localhost:{random_port}', asyncio=True)
+
+    async def _request(stream_param):
+        return [
+            response
+            async for response in client.post(
+                '/',
+                random_docs(1),
+                request_size=1,
+                max_attempts=5,
+                on_always=on_always_mock,
+                on_error=on_error_mock,
+                on_done=on_done_mock,
+                return_responses=True,
+                timeout=0.5,
+                stream=stream_param,
+            )
+        ]
+
+    if protocol == 'grpc':
+        stream_opts = [True, False]
+        for stream_param in stream_opts:
+            with pytest.raises(ConnectionError):
+                await _request(stream_param)
+    else:
+        with pytest.raises(aiohttp.ClientConnectorError):
+            await _request(stream_param=True)
+
+
+@pytest.mark.timeout(90)
+@pytest.mark.parametrize('flow_or_deployment', ['deployment', 'flow'])
+def test_grpc_stream_transient_error(flow_or_deployment, port_generator, mocker):
+    random_port = port_generator()
+    cntx = (
+        Flow(port=random_port).add(uses=MyExec)
+        if flow_or_deployment == 'flow'
+        else Deployment(include_gateway=True, uses=MyExec, port=random_port)
+    )
+
+    client = Client(host=f'{cntx.protocol}://{cntx.host}:{random_port}')
+    stop_event = multiprocessing.Event()
+    t = multiprocessing.Process(target=_start_runtime, args=(cntx, stop_event))
+    t.start()
+
+    max_attempts = 5
+    initial_backoff = 0.8
+    backoff_multiplier = 1.5
+    max_backoff = 5
+    try:
+        for attempt in range(1, max_attempts + 1):
+            try:
+                on_error_mock = mocker.Mock()
+                response = client.post(
+                    '/',
+                    random_docs(1),
+                    request_size=1,
+                    on_error=on_error_mock,
+                    return_responses=True,
+                    timeout=0.5,
+                )
+                assert len(response) == 1
+
+                on_error_mock.assert_not_called()
+            except ConnectionError as err:
+                run_async(
+                    wait_or_raise_err,
+                    attempt=attempt,
+                    err=err,
+                    max_attempts=max_attempts,
+                    backoff_multiplier=backoff_multiplier,
+                    initial_backoff=initial_backoff,
+                    max_backoff=max_backoff,
+                )
+    finally:
+        stop_event.set()
+        t.join()
+        t.terminate()
+
+
+@pytest.mark.timeout(90)
+@pytest.mark.asyncio
+@pytest.mark.parametrize('flow_or_deployment', ['deployment', 'flow'])
+@pytest.mark.ignore
+async def test_async_grpc_stream_transient_error(
```

Comment:
@alaeddine-13 @JoanFM This test is ignored because it creates a segmentation fault. I need some help to find the root cause.

Reply:
Looking at the logs, the warmup tasks plus the client retires might be causing an issue with the grpc asyncio implementation.

Reply:
I think we should remove `warmup tasks`